src/common/updating/dotnet_installation.cpp

```diff
@@ -0,0 +1,64 @@
+#include "pch.h"
+
+#include <common/common.h>
+
+#include "http_client.h"
+#include "dotnet_installation.h"
+
+namespace fs = std::filesystem;
+
+namespace updating
+{
+    bool dotnet_is_installed()
+    {
+        auto runtimes = exec_and_read_output(LR"(dotnet --list-runtimes)");
+        if (!runtimes)
+        {
+            return false;
+        }
+        const char DESKTOP_DOTNET_RUNTIME_STRING[] = "Microsoft.WindowsDesktop.App 3.1.";
+        return runtimes->find(DESKTOP_DOTNET_RUNTIME_STRING) != std::string::npos;
+    }
+
+    bool install_dotnet()
+    {
+        const wchar_t DOTNET_DESKTOP_DOWNLOAD_LINK[] = L"https://download.visualstudio.microsoft.com/download/pr/d8cf1fe3-21c2-4baf-988f-f0152996135e/0c00b94713ee93e7ad5b4f82e2b86607/windowsdesktop-runtime-3.1.4-win-x64.exe";
```

Comment:
The current version is 3.1.6, any particular reason for not installing the latest?